app/src/main/java/me/guillaumin/android/osmtracker/gpx/ExportAndShareTask.java

```diff
@@ -0,0 +1,168 @@
+     *
+     * @return
+     */
+    @Override
+    protected boolean updateExportDate() {
+        return true;
+    }
+
+    @Override
+    protected void onPostExecute(Boolean success) {
+        super.onPostExecute(success);
+
+        File cacheDir = context.getCacheDir();
+        File fileToShare;
+        File[] generatedFiles = trackDir.listFiles();
+        boolean a;
+        if (generatedFiles.length == 1) {
+            // if there's only one file in the directory - move file to cache root and share it
+            fileToShare = new File(cacheDir, "track-to-share.gpx");
+            int i = 0;
+            while (fileToShare.exists())
+                fileToShare = new File(cacheDir, "track-to-share" + i++ + ".gpx");
+            if (generatedFiles[0].renameTo(fileToShare))
+                a = trackDir.delete();
+            else
+                // in some strange situation, when file can't be moved
+                fileToShare = generatedFiles[0];
+        }
+        else if (generatedFiles.length > 1) {
+            // if there are several files - put them into zip and share it
+            try {
+                if (filename.endsWith(".gpx"))
+                    filename = filename.substring(0, filename.length() - 4);
+                filename = filename + ".zip";
+                fileToShare = File.createTempFile("track-to-share", ".zip", cacheDir);
+                BufferedInputStream bufferedInputStream;
+                byte[] buffer = new byte[4096];
+                int count;
+                FileOutputStream outputStream = new FileOutputStream(fileToShare);
+                ZipOutputStream zipOut = new ZipOutputStream(new BufferedOutputStream(outputStream));
+                for (File f : generatedFiles) {
+                    ZipEntry entry = new ZipEntry(f.getName());
+                    zipOut.putNextEntry(entry);
+                    bufferedInputStream = new BufferedInputStream(new FileInputStream(f));
+                    while ((count = bufferedInputStream.read(buffer, 0, buffer.length)) > -1)
+                        zipOut.write(buffer, 0, count);
+                    bufferedInputStream.close();
+                    f.delete();
+                }
+                zipOut.close();
+                trackDir.delete();
+            }
+            catch (IOException ioe) {
+                Log.e(TAG, "Can't add entry to zip file", ioe);
+                throw new IllegalStateException("Can't add entry to zip file", ioe);
+            }
+        }
+        else {
+            Log.e(TAG, "There's no files in track directory");
+            throw new IllegalStateException("There's no files in track directory");
+        }
+        sharedFiles.put(filename, fileToShare);
+
+        Uri tempFileUri = Uri.parse("content://me.guillaumin.android.osmtracker.fileshareprovider/" + filename);
+        Intent shareTrack = new Intent(Intent.ACTION_SEND);
+        shareTrack.setType("application/xml");
```

Comment:
My understanding is that may also be a ZIP file, not always an XML file? I guess that should be reflected in the type here?